ethpm/backends/ipfs.py

```diff
@@ -0,0 +1,104 @@
+from eth_utils import to_bytes
+import requests
+
+from ethpm import V2_PACKAGES_DIR
+from ethpm.backends.base import BaseURIBackend
+from ethpm.constants import INFURA_GATEWAY_PREFIX, IPFS_GATEWAY_PREFIX
+from ethpm.utils.ipfs import extract_ipfs_path_from_uri, is_ipfs_uri
+
+
+class BaseIPFSBackend(BaseURIBackend):
+    """
+    Base class for all URIs with an IPFS scheme.
+    """
+
+    def can_handle_uri(self, uri: str) -> bool:
+        """
+        Return a bool indicating whether or not this backend
+        is capable of serving the content at the URI.
+        """
+        return is_ipfs_uri(uri)
+
+    def fetch_uri_contents(self, uri: str) -> bytes:
+        raise NotImplementedError("Must be implemented by subclass")
+
+    @property
+    def base_uri(self) -> str:
+        raise NotImplementedError("Must be implemented by subclass")
+
+
+class IPFSOverHTTPBackend(BaseIPFSBackend):
```

Comment:
I think you can just put an abstract `base_uri` method on this class to take Piper's suggestion of having `base_uri` on an abstract class for HTTP backends.